### PR TITLE
fix(argo-workflows): Fix controller Role/ClusterRole to work with InstanceID

### PR DESCRIPTION
**Pull request by kobejn-jb**
Signed-off-by: kobejn-jb jakubbielawski89@gmail.com

This PR comes with the following changes:

    It allows for the controller to work with InstanceID while using leases
    It fixes a controller Role/ClusterRole, now it allows to read and update leases with names ending with -<your instance ID> when controller is set to use InstanceID

This PR intends to solve the following use case. In case when controller is set to use InstanceID controller role does not allow to get/watch/update/path/delete leases with names different from "workflow-controller-lease" and "workflow-controller-lease" but controller tries to use lease named "workflow-controller-lease-<provided InstanceID>". Now trying to use controller with InstanceID it results with error:

`E0713 13:43:58.014735       1 leaderelection.go:325] error retrieving resource lock argo-test/workflow-controller-argo-controller: leases.coordination.k8s.io "workflow-controller-argo-controller" is forbidden: User "system:serviceaccount:argo-test:argo" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "argo-test"`

This PR changes Role/ClusterRole template, when InstanceID is enabled names of leases listed in permissions in Roles will contain that name at the end

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [X] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [X] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-workflows/Chart.yaml [hunk 1]
```diff
@@ -16,3 +16,4 @@ maintainers:
 annotations:
   artifacthub.io/changes: |
     - "[Added]: Initialize Changelog"
+    - "[Fixed]: Controller ClusterRole to work with InstanceID"
```
Comment (vladlosev) [inline charts/argo-workflows/Chart.yaml:19]: As I understand from https://artifacthub.io/docs/topics/annotations/helm/, you should replace the old value of the annotation, not append to it.

File: charts/argo-workflows/templates/controller/workflow-controller-cluster-roles.yaml [hunk 2]
```diff
@@ -121,8 +121,18 @@ rules:
   resources:
   - leases
   resourceNames:
+  {{- if .Values.controller.instanceID.enabled }}
+  {{- if .Values.controller.instanceID.useReleaseName }}
```
Comment (vladlosev) [inline charts/argo-workflows/templates/controller/workflow-controller-cluster-roles.yaml:125]: Please indent the template flow control tags. It makes the code easier to read and is consistent with the rest of the file.

File: charts/argo-workflows/Chart.yaml (hunk 1 above)
Comment (kobejn-jb) [reply charts/argo-workflows/Chart.yaml]: Previously I've misunderstood how it works, fixed.

File: charts/argo-workflows/templates/controller/workflow-controller-cluster-roles.yaml (hunk 2 above)
Comment (kobejn-jb) [reply charts/argo-workflows/templates/controller/workflow-controller-cluster-roles.yaml]: I've added indent but there is very similar flow control in configMap template:
https://github.com/argoproj/argo-helm/blob/922799081d6037d71971baf3fecf3519c35326e9/charts/argo-workflows/templates/controller/workflow-controller-config-map.yaml#L9-L11
Should I also change it to make files more consistent?

Comment (vladlosev) [reply charts/argo-workflows/templates/controller/workflow-controller-cluster-roles.yaml]: That would be nice, please do.

Comment (kobejn-jb) [reply charts/argo-workflows/templates/controller/workflow-controller-cluster-roles.yaml]: Done

**Review by vladlosev — APPROVED** [review]
LGTM

**Comment by kobejn-jb** [comment]
What is the procedure here? When can I expect this to be merged and released?

**Pull request merged**